grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/cache/Cache.java

```diff
@@ -139,24 +137,14 @@ public void add(Cache<Q, T> c2){
      */
     public abstract Stream<Answer> record(Q query, Stream<Answer> answers);
 
-    /**
-     * record answer stream for a specific query and retrieve the updated stream in a lazy iterator
-     * @param query to be recorded
-     * @param answers answer stream of the query
-     * @return lazy iterator of updated answers
-     */
-    public abstract LazyIterator<Answer> recordRetrieveLazy(Q query, Stream<Answer> answers);
-
     /**
      * retrieve (possibly) cached answers for provided query
      * @param query for which to retrieve answers
      * @return unified cached answers
      */
     public abstract T getAnswers(Q query);
-    public abstract Pair<T, MultiUnifier> getAnswersWithUnifier(Q query);
+
     public abstract Stream<Answer> getAnswerStream(Q query);
-    public abstract Pair<Stream<Answer>, MultiUnifier> getAnswerStreamWithUnifier(Q query);
-    public abstract LazyIterator<Answer> getAnswerIterator(Q query);
```

Comment:
why are we removing them? both the `QueryCache` and `LazyQueryCache` implement these methods

Reply:
IntelliJ thinks it isn't used because everything that use it is using the method on `QueryCache` or `LazyQueryCache` directly, so the interface method isn't needed.

I don't know if that's a very good reason to remove it though... I'll probably put it back!

Reply:
yeah I would put them back :)